packages/react-native/ReactCommon/react/nativemodule/core/platform/android/ReactCommon/JavaTurboModule.cpp

```diff
@@ -801,18 +920,30 @@ jsi::Value JavaTurboModule::invokeJavaMethod(
               throw std::invalid_argument("Promise fn arg count must be 2");
             }
 
+            // JS Stack at the time when the promise is created.
+            std::optional<std::string> jsInvocationStack;
+            if (isSavePromiseJSInvocationStackEnabled()) {
+                jsInvocationStack = createJSRuntimeError(runtime, "")
+                  .asObject(runtime)
+                  .getProperty(runtime, "stack")
+                  .asString(runtime)
+                  .utf8(runtime);
```

Comment:
Could we somehow keep this as a jsi::Value, and pass that to `createPromiseRejectJavaCallbackFromJSIFunction` so have the "real" stack property, without parsing to/from string?

Reply:
`std::optional<std::reference_wrapper<jsi::Value>>` seems to work

Reply:
I didn't realize this was just a string. Probably simpler from a memory model to keep this as string (which is also what you have on iOS) so we don't need to worry about this jsi::value outliving the runtime.

Reply:
I reverted the change to `jsi::Value` back to `string`